### PR TITLE
Fix: Update static posters flag default value to true

### LINKED ISSUE #22977: ReaderStreamViewController crash on WordPress app
Sentry Issue: [WORDPRESS-IOS-48BX](https://a8c.sentry.io/issues/5131166213/?referrer=github_integration)

```
EXC_BREAKPOINT: Exception 6, Code 1, Subcode 4341699504
  File "<compiler-generated>", in range
  File "<compiler-generated>", in Array._checkSubscript
  File "<compiler-generated>", in Array.subscript.getter
  File "ReaderStreamViewController.swift", line 1592, in ReaderStreamViewController.tableView
  File "ReaderStreamViewController.swift", line 1592, in ReaderStreamViewController.tableView
...
(55 additional frame(s) were not displayed)
```

### PR DESCRIPTION
**Pull request by dvdchr**
Fixes #22977 
Internal ref p1712651805548069-slack-C012H19SZQ8

This issue affects the WordPress app. Due to the JP feature removal flags' default value set to `false`, when the user fails to fetch the remote feature flag values, [they will get `.normal` value from the JetpackFeaturesRemovalCoordinator](https://github.com/wordpress-mobile/WordPress-iOS/blob/febe218807e1f843691c44ad0dab830f0faa78a3/WordPress/Classes/ViewRelated/Jetpack/Branding/Coordinator/JetpackFeaturesRemovalCoordinator.swift#L93).

This PR solves that by updating the static poster flag's default value to `true`, which should prevent WP app users from unintendedly accessing the Reader (or any other Jetpack feature). The decision on picking which flag's default value to set to `true` is based on which flag has the most recent app version requirement, based on the internal implementation here: fbhepr%2Skers%2Sjcpbz%2Sjc%2Qpbagrag%2Syvo%2Ssrngher%2Qsyntf%2Spbasvt%2Szbovyr%2Qnccf.cuc%3Se%3Q96953s75%26zb%3Q990%26sv%3Q36%2333%2Q49-og

h/t to @staskus for finding out the root cause! Also, I'm adding @hassaanelgarem as an extra pair of eyes to verify if updating the default value of the static posters phase is the right move here.

---

<details>

<summary>Previous outdated description</summary>

~~This PR adds a workaround for an index out-of-bound exception when the table view tries to get a post object from a collection of posts obtained from a fetched results controller. This particular part of the code remains unchanged, but the Reader hierarchy was [recently changed in 24.2](https://github.com/wordpress-mobile/WordPress-iOS/blob/9fb2da734f2591f2311264217b030665430885e0/RELEASE-NOTES.txt#L44), when we introduced a new UI for the Reader stream. This changes the structure from tabbed view (e.g., load 5 tabs at once) to just a single stream where users could switch streams and trigger content reloads.~~

~~I'm still unsure why this crash first happened in 24.5, since I don't think there are any major changes [between 24.4 and 24.5](https://github.com/wordpress-mobile/WordPress-iOS/compare/24.4...24.5) that would introduce a race condition for the table view.~~

</details>

## To test

**EDIT**: Testing steps updated based on @staskus' findings in https://github.com/wordpress-mobile/WordPress-iOS/pull/22991#discussion_r1560790955:

### WordPress app

- Launch the WordPress app and log in to the app.
- Force close the app, and turn on airplane mode.
- Launch WordPress app again.
- 🔎 Verify that the Reader tab is NOT accessible.

### Jetpack app

- Reader
- Search for `松,`
- Scroll down until more content loads and `reader_infinite_scroll_performed` is logged
- Select input field
- Tap "Done"
- 🔎 Verify that the app does not crash

## Regression Notes
1. Potential unintended areas of impact
Should be none. Out of bound indexes shouldn't crash

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manually tested the changes. Relied on existing tests in `JetpackFeaturesRemovalCoordinatorTests`.

3. What automated tests I added (or what prevented me from doing so)
Added unit tests to cover offline cases for WordPress and Jetpack app.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Testing checklist:
N/A — not a UI change.

- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/wordpress-mobile/WordPress-iOS/milestone/278">24.7</a>. This milestone is due in less than 4 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WordPress/Classes/ViewRelated/Reader/ReaderStreamViewController.swift [hunk 1]
```diff
@@ -1585,12 +1585,11 @@ extension ReaderStreamViewController: WPTableViewHandlerDelegate {
     }
 
     func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
-        guard let posts = content.content as? [ReaderPost] else {
+        guard let posts = content.content as? [ReaderPost],
+              let post = posts[safe: indexPath.row] else {
```
Comment (staskus) [inline WordPress/Classes/ViewRelated/Reader/ReaderStreamViewController.swift:1590]: As we spoke, the issue happened on the WordPress app when Reader appeared after remote feature flags could not be fetched.

I managed to reproduce the issue following the steps of this bug report https://a8c.sentry.io/issues/5131166213/events/871dd2f9e0b04b9b83cc1c84ef73f506/?project=1438083&referrer=previous-event

I noticed that all the logs contain `reader_infinite_scroll_performed`. The steps I used to reproduce the issue:

1. Reader
2. Search for `松,`
3. Scroll down until more content loads and `reader_infinite_scroll_performed` is logged
4. Select input field
5. Tap "Done"

Comment (staskus) [reply WordPress/Classes/ViewRelated/Reader/ReaderStreamViewController.swift:1590]: This fix helps to avoid the crash but it may be worth understanding why the issue happens.

**Review by staskus — APPROVED** [review]
Thanks for updating the solution with tests! 

It's a weird collection of corner cases that caused this issue. Crashes shouldn't be happening anymore, and maybe not presenting the Reader on WP app won't even trigger the condition in the first place. We'll be able to see that through new logs.
